src/rc_allocator.c

```diff
@@ -236,6 +236,39 @@ rc_allocator_config_init(rc_allocator_config *allocator_cfg,
    allocator_cfg->extent_capacity = capacity / io_cfg->extent_size;
 }
 
+/*
+ *----------------------------------------------------------------------
+ * rc_allocator_valid_config() --
+ *
+ * Do minimal validation of RC-allocator cofiguration.
+ *----------------------------------------------------------------------
+ */
+platform_status
+rc_allocator_valid_config(rc_allocator_config *cfg)
+{
+   platform_status rc = STATUS_OK;
+   rc                 = laio_config_valid(cfg->io_cfg);
+   if (!SUCCESS(rc)) {
+      return rc;
+   }
+
+   if (cfg->capacity == 0) {
+      platform_error_log("Configured disk size %lu bytes is invalid.\n",
+                         cfg->capacity);
+      return STATUS_BAD_PARAM;
+   }
+   if (cfg->extent_capacity == 0) {
+      platform_error_log("Configured extent capacity %lu bytes is invalid.\n",
+                         cfg->extent_capacity);
+      return STATUS_BAD_PARAM;
+   }
+   if (cfg->capacity != (cfg->io_cfg->extent_size * cfg->extent_capacity)) {
+      return STATUS_BAD_PARAM;
+   }
+   return rc;
+}
```

Comment:
This was an error handling issue I ran into while building the sample programs, where some input config-params had the wrong units.

I think these are "small" enough that we can roll these in as part of this examples-PR.

If there is a strong desire to check these in separately, let me know. I can peel these off to a separate commit and integrate them first.

Reply:
(Have reverted these diffs, so that this change-set reduces to just the C-sample programs. Ignore this diff.)

Reply:
Update: Above error condition checks have been separately merged under commit SHA e677a4e.